packages/console/src/components/Checkbox/Icon.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+
+type Props = {
+  className?: string;
+};
+
+const Icon = ({ className }: Props) => (
+  <span className={className}>
+    <svg
+      width="20"
+      height="20"
+      viewBox="0 0 20 20"
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
+      className={className}
+    >
+      <rect x="1" y="1" width="18" height="18" rx="4" fill="currentColor" />
+      <path
+        fillRule="evenodd"
+        clipRule="evenodd"
+        d="M8.31476 13.858L5.13295 10.441C4.95568 10.253 4.95568 9.947 5.13295 9.757L5.77568 9.074C5.95295 8.886 6.24113 8.886 6.4184 9.074L8.63657 11.466L13.5811 6.141C13.7584 5.953 14.0465 5.953 14.2238 6.141L14.8665 6.825C15.0438 7.013 15.0438 7.32 14.8665 7.507L8.95748 13.858C8.78021 14.046 8.49203 14.046 8.31476 13.858Z"
+        fill="white"
+      />
+    </svg>
+    <svg
+      width="20"
+      height="20"
+      viewBox="0 0 20 20"
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
+      className={className}
+    >
+      <rect x="2" y="2" width="16" height="16" rx="3" stroke="currentColor" strokeWidth="2" />
+    </svg>
+    <svg
+      width="20"
+      height="20"
+      viewBox="0 0 20 20"
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
+      className={className}
+    >
+      <rect x="1" y="1" width="18" height="18" rx="4" fill="currentColor" />
+      <path
+        fillRule="evenodd"
+        clipRule="evenodd"
+        d="M8.31476 13.858L5.13295 10.441C4.95568 10.253 4.95568 9.947 5.13295 9.757L5.77568 9.074C5.95295 8.886 6.24113 8.886 6.4184 9.074L8.63657 11.466L13.5811 6.141C13.7584 5.953 14.0465 5.953 14.2238 6.141L14.8665 6.825C15.0438 7.013 15.0438 7.32 14.8665 7.507L8.95748 13.858C8.78021 14.046 8.49203 14.046 8.31476 13.858Z"
+        fill="white"
+      />
+    </svg>
+    <svg
+      width="20"
+      height="20"
+      viewBox="0 0 20 20"
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
+      className={className}
+    >
+      <rect
+        x="2"
+        y="2"
+        width="16"
+        height="16"
+        rx="3"
+        fill="#EFF1F1"
+        stroke="currentColor"
+        strokeWidth="2"
+      />
+    </svg>
+  </span>
```

Comment:
If you look at the svgs carefully you'll see their containers are exactly the same, which means we don't have to replicate it 4 times.
Also using class to conditionally show/hide the svgs seems a bit weird.
You can treat the svgs as regular components, and instead of just using classname, you can define `checked: boolean` and `disabled: boolean` as props of Icon component. And inside, use one svg container and render the `rect` and `path` conditionally by the `checked` or `disabled` props

Reply:
You can even import styles and use css class to set the fill or stroke colors in the Icon component.

Reply:
Is not easy to get the right `checked` prop, as it can be changed with `ref`, that would happen when use use `register` in `react-hook-form`. If we have to implement by this way, need to change the props into `value` and `onChange` pair, but I think for this simple component, we can keep it behaves like the html's native `input`.

That's why I use css selectors, make it as "native" as possible, that would help if we want to do `a11y` in the future.

Reply:
Just as `disabled`, the input element should also have a `checked` prop. Would you check on that?

Reply:
Per offline discussion, is not easy to get the right value, leave if for now.